src/connection_wrap.h

```diff
@@ -0,0 +1,36 @@
+#ifndef SRC_CONNECTION_WRAP_H_
+#define SRC_CONNECTION_WRAP_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "async-wrap.h"
```

Comment:
I’d read `other headers it needs` as not necessarily including headers that this header already has a transitive dependency on, like `async-wrap.h`, as `AsyncWrap` must already be (at least forward-)declared in `stream_wrap.h` to be used in the constructor of `StreamWrap`.

It probably doesn’t really matter, and if you want to keep it this way for clarity, that works for me.